### PR TITLE
[CI] Refactor sycl-containers to use matrix

### PR DESCRIPTION
**Pull request by jsji**
Address review comments from https://github.com/intel/llvm/pull/13272#discussion_r1552471814

### CODE REVIEW

**Review by aelovikov-intel — APPROVED** [review]

**Comment by jsji** [comment]
@intel/llvm-gatekeepers Please help to merge . Thanks.

File: .github/workflows/sycl-containers.yaml [hunk 1]
```diff
@@ -24,12 +24,31 @@ on:
...
+            file: ubuntu2204_build
+            tags: ubuntu2204_build
+            build_args: ""
+          - name: Intel Drivers Ubuntu 22.04 Docker image
+            file: ubuntu2204_intel_drivers
+            tags: ubuntu2204_intel_drivers
+            build_args: "use_latest=false"
+          - name: Intel Drivers (unstable) Ubuntu 22.04 Docker image
+            file: ubuntu2204_intel_drivers
+            tags: ubuntu2204_intel_drivers_unstable
```
Comment (bader) [inline .github/workflows/sycl-containers.yaml:50]: I think this patch changes tags we used for uploading docker images. 
@jsji, could you update the documentation https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/DockerBKMs.md#sycl-containers-overview if it's needed, please?

Comment (jsji) [reply .github/workflows/sycl-containers.yaml:50]: Yes, the tag for `unstable` is changed, but we never mentioned/use it anywhere in source tree/docs. So I don't think any updated needed. Let me know if there is some usage that I don't know.

Comment (jsji) [reply .github/workflows/sycl-containers.yaml:50]: https://github.com/intel/llvm/pull/13307 for the image update first.

**Pull request merged**